homeassistant/components/switcher_kis/cover.py

```diff
@@ -72,7 +83,15 @@ def __init__(
         super().__init__(coordinator)
         self._cover_id = cover_id
 
-        self._attr_unique_id = f"{coordinator.device_id}-{coordinator.mac_address}"
+        if (
```

Comment:
It would be better to have a base class with the common functionality and two subclasses, similar to:
https://github.com/home-assistant/core/blob/e54031e318ff7af281ae399d6de9743411bf31a9/homeassistant/components/switcher_kis/switch.py#L80

So you would end up with
- `SwitcherBaseCoverEntity`
- `SwitcherSingleCoverEntity`
- `SwitcherDualCoverEntity`

The subclass will only implement changed parts

Reply:
It is done.. I tried to separate __init__ as well but it did make issues..

Reply:
Please submit the separate init and we will fix the issues in the PR review.

Reply:
It is done. You can see current issue.